charmhelpers/osplatform.py

```diff
@@ -9,9 +10,13 @@ def get_platform():
     This string is used to decide which platform module should be imported.
     """
     # linux_distribution is deprecated and will be removed in Python 3.7
-    # Warings *not* disabled, as we certainly need to fix this.
-    tuple_platform = platform.linux_distribution()
-    current_platform = tuple_platform[0]
+    # Warnings *not* disabled, as we certainly need to fix this.
+    if hasattr(platform, 'linux_distribution'):
+        tuple_platform = platform.linux_distribution()
+        current_platform = tuple_platform[0]
+    else:
+        current_platform = _get_platform_from_fs()
```

Comment:
I think you can get these details from:
import distro
distro.linux_distribution()

Reply:
question for the larger community... i didn't see a "requirements.txt" but i did set a setup.py.  I'd be willing to add the 'distro' package as a dependent library but didn't know the communities reaction to addition of another library and if the downstream charms would somehow know to add this dep.

Reply:
That is a good question.

Reply:
Is `distro` on PyPi and pure python?  If not, then it wouldn't work in a venv, which would break reactive charms (and maybe the new operator framework).

Reply:
After more discussion I think what you have here is good, as opposed to requiring a new dependency to be pulled into classic charms. I tried your code out successfully and it looks good. Do you know if this works with CentOS?

Reply:
`distro` is an extra package.  It would force classic charms to include it.  That's why this implementation was chosen.

```
(venv)root@first:~/juju/charm-helpers# python
Python 3.4.3 (default, Nov 12 2018, 22:25:49) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import distro
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'distro'
```

Reply:
i'll try to run a lxd container in centos w/ py38 and ensure this works

Reply:
OK... i took a simpler approach.  the answer is yes, it works on [centos](https://paste.ubuntu.com/p/2KkVb85T8v/)

Reply:
thanks for checking